c7n/resources/appmesh.py

```diff
@@ -0,0 +1,280 @@
+        # do not use this approach.  App mesh list_meshes doesn't
+        # support filtering.  ...
+        #
+        # https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/appmesh/client/list_meshes.html  # noqa
+        #
+        # and so when an event is received then the enum function gets
+        # called and the event id's get enriched.
+
+        # for example the specific identity found in an
+        # event. However, if the enum op doesn't support filtering
+        # then what will happen with events instead is a full list of
+        # resources followed by client side filtering.
+        #
+        # params:
+        #  enum_op - the aws api op
+        #  path - JMESPATH path to the field in the response that is the collection of result
+        #         objects
+        #  extra_args - eg {'maxResults': 100}
+        #
+        enum_spec = ('list_meshes', 'meshes', None)
+
+        # In many cases the enum_spec function is one of the
+        # "describe_" style functions that return a full'ish spec that
+        # is sufficient for the user detection, however in those cases
+        # where the enum_spec is a "list_" style funtion then the
+        # response to then enum call will be lacking in detail and
+        # might just be a list of id's. In these cases it is generally
+        # necessary to define a "detail_spec" which can be used to
+        # enrich the values provided by the enum_spec.
+
+        # detail_op = boto api call name
+        # param_name = name of argument to boto api call
+        # param_key = name of field in enum_spec response to drive this call
+        # detail_path = path to pull out of the boto response and
+        #               return as the detail result if not provided
+        #               then whole response is included in results
+        detail_spec = ('describe_mesh', 'meshName', 'meshName', None)
+
+        # This function gets called with the id's retrieved from the
+        # event and is expected to return the enriched resource
+        # definition.
+        #
+        # Explanation of logic:
+        #
+        # The default impl on the superclass calls the enum_spec
+        # function, however for many resource types this is useless
+        # since the enum function provides little specification and
+        # often everything the enum function provides is also on the
+        # detail_spec function.
+        #
+        # So where we're going to have to have a detail_spec and where
+        # the enum_spec is redundant then why call the enum spec
+        # function at all?  This is a small speed / cost saving as it
+        # reduces the number of AWS API calls made for each event.
+        # So, in the the case of the current resource we will simply
+        # skip the enum call entirely.
+        #
+        # TODO: Propose an improvement where we allow skipping of the
+        # enum_spec call self.source.get_resources(ids)

```

Comment:
a source subclass is free to do that, normally this is not in a query manager, but a describe source subclass, you can grep `DescribeSource` for extant examples .

Reply:
I removed my logic as it meant the pull and event models differred.